src/utils/LoggerManager.cpp

```diff
@@ -0,0 +1,45 @@
+#include "LoggerManager.h"
+#include <vector>
+#include <spdlog/sinks/stdout_color_sinks.h>
+#include <spdlog/sinks/basic_file_sink.h>
+#include <spdlog/async.h>
+
+/**
+ * Sets the logging level and ensures immediate flushing at this level.
+ *
+ * @param level The log level to be set for the logger.
+ */
+void LoggerManager::setLogLevel(spdlog::level::level_enum level) {
+    auto logger = LoggerManager::getLogger();
+    logger->set_level(level);
+    logger->flush_on(level);
+}
```

Comment:
I don't know how i feel about this one. 
This probably don't need to be it's own function and instead be part of the initialization of the Logger or make it a private function.